src/rust/engine/src/scheduler.rs

```diff
@@ -291,11 +292,33 @@ impl Scheduler {
       .map(|s| s.into())
       .collect();
 
+    // This map keeps the k most relevant jobs in assigned possitions.
+    // Keys are positions in the display (display workers) and the values are the actual jobs to print.
+    let mut tasks_to_display = IndexMap::new();
+
     let results = loop {
       if let Ok(res) = receiver.recv_timeout(Duration::from_millis(100)) {
         break res;
       } else if let Some(display) = optional_display.as_mut() {
-        Scheduler::display_ongoing_tasks(&self.core.graph, &roots, display);
+        // Update the graph. To do that, we iterate over heavy hitters.
+        let mut heavy_hitters: Vec<(String, Duration)> = self
+          .core
+          .graph
+          .heavy_hitters(&roots, display.worker_count());
+        // Insert every one in the set of tasks to display.
+        for (task, duration) in &heavy_hitters {
+          // TODO I don't really want to trigger the clone unconditionally, but using an
```

Comment:
Yea, folks have complained about the hashmap `Entry` API in this regard... keys get cloned quite a bit.

There isn't really a non-awkward solution, so probably fine to just remove the TODO here.